src/shared/libfido2-util.c

```diff
@@ -519,6 +638,30 @@ int fido2_use_hmac_hash(
                         goto finish;
                 }
 
+                r = fido2_is_cred_in_specific_token(path, rp_id, cid, cid_size, pins, required);
+                /* We handle PIN related errors here since we have used PIN in the check.
+                 * If PIN error happens, we can avoid pin retries decreased the second time. */
+                if (IN_SET(r, -ENOANO, -ENOLCK, -EOWNERDEAD)) {
```

Comment:
maybe write this as:

```c
if (IN_SET(r, -ENOANO,        /* → pin required */
              -ENOLCK,        /* → pin incorrect */
              -EOWNERDEAD)) { /* → verification (uv) blocked */
```

or so. i.e. we use unix error codes here, and they are really cryptic, hence it's always good to explain what an error we handle specifically means where we handle it.